crane/hooks/slack.py

```diff
@@ -209,54 +211,58 @@ def generate_env_lines(self, env_lines, status):
             env_lines.append(status + " " + self.env_text)
 
     def before_upgrade(self):
-        message = self.get_existing_message() or self.generate_new_message()
-        fields = message["attachments"][0]["fields"]
+        messages = self.get_existing_messages() or [self.generate_new_message()] * len(self.channel_ids)
 
-        if "ts" in message:
-            self.send_reply(message["ts"], f"Starting release on {self.env_text}.")
+        for channel_id, message in zip(self.channel_ids, messages):
```

Comment:
Or would it somehow be possible to keep channel id in message?

Reply:
Kind of. I refactored the function to:

```python
    def get_existing_messages(self):
        return {
            channel_id: self.get_existing_message(channel_id)
            for channel_id in self.channel_ids
        }
```

So usage is now simply just:

```python
        for channel_id, message in messages.items():
```